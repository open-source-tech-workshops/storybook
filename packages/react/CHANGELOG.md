# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.2.1](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.2.0...v0.2.1) (2023-01-01)

**Note:** Version bump only for package @twk-ui-lib/react

# [0.2.0](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.1.0...v0.2.0) (2022-12-30)

### Styles

- rate style change ([6f5a091](https://github.com/twk-online-shopping-platform/twk-ui-library/commit/6f5a091e4a1a3c45b3e03a77b07c6c3842343046))

### BREAKING CHANGES

- rate ui component ui style could be affected

# [0.1.0](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.9...v0.1.0) (2022-12-30)

### Styles

- direct style import ([0df8afa](https://github.com/twk-online-shopping-platform/twk-ui-library/commit/0df8afa522677d2294c88bf0aaf94dd9556fe5e7))

### BREAKING CHANGES

- style display for ui component could be affected

## [0.0.9](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.8...v0.0.9) (2022-12-30)

### Bug Fixes

- automate publish auth ([931a74f](https://github.com/twk-online-shopping-platform/twk-ui-library/commit/931a74fde28de151dbbe50c1258048fa803069e1))

## [0.0.8](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.7...v0.0.8) (2022-12-30)

### Bug Fixes

- remove opt ([57cc5a8](https://github.com/twk-online-shopping-platform/twk-ui-library/commit/57cc5a82de018a91a9408cb07cf4a5d095e8423a))

## [0.0.7](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.1...v0.0.7) (2022-12-30)

**Note:** Version bump only for package @twk-ui-lib/react

## [0.0.6](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.1...v0.0.6) (2022-12-30)

**Note:** Version bump only for package @twk-ui-lib/react

## [0.0.5](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.1...v0.0.5) (2022-12-25)

**Note:** Version bump only for package @twk-ui-lib/react

## [0.0.4](https://github.com/twk-online-shopping-platform/twk-ui-library/compare/v0.0.3...v0.0.4) (2022-12-17)

### Bug Fixes

- **jest.config.js:** jest config fix ([f3b7c3d](https://github.com/twk-online-shopping-platform/twk-ui-library/commit/f3b7c3dfcffcf9530083ac4ec18237b6b04ae7c3))
