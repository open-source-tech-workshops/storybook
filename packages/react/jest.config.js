module.exports = {
  roots: ["<rootDir>/src"],
  testRegex: "((\\.|/*.)(spec))\\.tsx?$",
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["<rootDir>/jest.setup.ts"],
  transform: {
    "^.+\\.tsx?$": "ts-jest",
    "node_modules/variables/.+\\.(j|t)sx?$": "babel-jest",
    "^.+\\.js$": "babel-jest",
    "\\.(css|less|sass|scss)$": "jest-css-modules-transform",
  },
  transformIgnorePatterns: ["<rootDir>/node_modules/blurhash/"],
};
