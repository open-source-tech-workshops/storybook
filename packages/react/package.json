{
  "name": "@twk-ui-lib/react",
  "version": "0.3.7",
  "description": "this is ui library to be used by micro front end",
  "main": "lib/index.js",
  "repository": "https://github.com/twk-online-shopping-platform/twk-ui-library.git",
  "homepage": "https://twk-ui-lib.netlify.app/",
  "author": "kina",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.20.5",
    "@babel/preset-react": "^7.18.6",
    "@storybook/addon-actions": "^6.5.13",
    "@storybook/addon-controls": "^6.5.13",
    "@storybook/addon-essentials": "^6.5.13",
    "@storybook/addon-interactions": "^6.5.13",
    "@storybook/addon-links": "^6.5.13",
    "@storybook/addon-postcss": "^2.0.0",
    "@storybook/builder-webpack4": "^6.5.13",
    "@storybook/manager-webpack4": "^6.5.13",
    "@storybook/react": "^6.5.14",
    "@storybook/testing-library": "^0.0.13",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.0.3",
    "@types/react": "^18.0.20",
    "@types/react-test-renderer": "^18.0.0",
    "babel-loader": "^9.1.0",
    "chromatic": "^6.13.1",
    "jest": "^29.1.1",
    "jest-canvas-mock": "^2.4.0",
    "jest-environment-jsdom": "^29.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-test-renderer": "^18.2.0",
    "rollup": "^3.7.4",
    "rollup-plugin-typescript2": "^0.34.1",
    "ts-jest": "^29.0.2",
    "typescript": "^4.8.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "files": [
    "lib"
  ],
  "scripts": {
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook",
    "test": "jest --coverage",
    "test:watch": "jest --verbose false --watch --maxWorkers=20%",
    "build": "rollup -c",
    "chromatic": "chromatic --project-token=${CHROMATIC_PROJECT_TOKEN}",
    "chromatic:check": "chromatic test --project-token=${CHROMATIC_PROJECT_TOKEN}"
  },
  "workspaces": [
    "packages/*"
  ],
  "dependencies": {
    "@fortawesome/fontawesome-free": "^6.2.0",
    "@iconscout/unicons": "^4.0.1",
    "blurhash": "^2.0.3",
    "bootstrap-icons": "^1.9.1",
    "jest-css-modules-transform": "^4.4.2",
    "react-blurhash": "^0.2.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "gitHead": "9f8a20e1c25a66e4bc74f354b9e80c541ffd51f2"
}
