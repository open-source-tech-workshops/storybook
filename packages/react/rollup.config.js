const Ts = require("rollup-plugin-typescript2");

module.exports = {
  input: ["src/index.ts", "src/atoms/index.ts"],
  output: {
    dir: "lib",
    format: "esm",
    sourcemap: true,
    preserveModules: true,
  },
  plugins: [Ts()],
  external: ["react"],
};
