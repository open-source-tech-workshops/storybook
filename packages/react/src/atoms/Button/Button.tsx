import React from "react";
import { TypographyVariant, TypographyWeight } from "../Typography/Type";
import { ButtondRadius, ButtonVariant } from "./Type";
import Typography from "../Typography/Typography";
import { ButtonTestId } from "./ButtonConstants";
import { ButtonType, ButtonSize } from "./Type";
import Icon from "../Icon/Icon";

const Button = ({
  label = "button",
  clickHandler,
  variant = ButtonVariant.PRIMARY,
  size = ButtonSize.SMALL,
  radius = ButtondRadius.BOTH,
  leftIcon,
  rightIcon,
}: ButtonType) => {
  let leftIconClassName: string | undefined = `icn-lf icn-${size}-input-field `;
  let rightIconClassName:
    | string
    | undefined = `icn-rt icn-${size}-input-field `;
  const buttonClass = `btn-${variant} btn-${size} b-rd-${radius} b-rd-blue b-rd-thick`;
  if (typeof leftIcon === "string") {
    leftIconClassName = leftIconClassName.concat(leftIcon);
  } else if (typeof leftIcon == "boolean" && leftIcon) {
    leftIconClassName = leftIconClassName.concat(
      "fa-sharp fa-solid fa-angle-left"
    );
  } else {
    leftIconClassName = undefined;
  }
  if (typeof rightIcon === "string") {
    rightIconClassName = rightIconClassName.concat(rightIcon);
  } else if (typeof rightIcon === "boolean" && rightIcon) {
    rightIconClassName = rightIconClassName.concat("fa-solid fa-angle-right");
  } else {
    rightIconClassName = undefined;
  }
  return (
    <button
      title="Button"
      className={buttonClass}
      onClick={() => {
        if (clickHandler) clickHandler();
      }}
      data-testid={ButtonTestId}
    >
      {leftIconClassName ? <Icon cssValue={leftIconClassName} /> : null}

      <Typography
        text={label}
        variant={TypographyVariant.BUTTON}
        weight={TypographyWeight.SEMI_BOLD}
      />
      {rightIconClassName ? <Icon cssValue={rightIconClassName} /> : null}
    </button>
  );
};

export default Button;
