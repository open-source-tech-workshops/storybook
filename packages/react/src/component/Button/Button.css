.twk-button {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: #1a73e8;
  border-radius: 19px;
  padding: 17px 48px;
  border-color: #ffffff;
}

.twk-text {
  width: 65px;
  height: 22px;

  /* Button/semibold/1 */

  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 18px;
  line-height: 22px;
  /* identical to box height */

  display: flex;
  align-items: center;
  text-align: center;

  /* White/White */

  color: #ffffff;

  /* Inside auto layout */

  flex: none;
  order: 0;
  flex-grow: 0;
}

.twk-button-color-red {
  background-color: red;
}

.twk-button-color-blue {
  background-color: blue;
}

.twk-button-color-green {
  background-color: green;
}
