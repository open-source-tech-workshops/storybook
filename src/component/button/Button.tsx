import React from 'react';
import './button.css';
import '../../assets/css/common.css'

interface ButtonProps {
  bgColor?: 'blue' | 'white' | 'black' | 'red' | 'green';
  txtColor?: 'blue' | 'white' | 'black' | 'red' | 'green';
  size?: 'small' | 'medium' | 'large';
  label: string;
  onClick?: () => void;
}

export const Button = ({
  size = 'medium',
  bgColor,
  txtColor,
  label,
  ...props
}: ButtonProps) => {
  const background: string = ['open-proj-btn',bgColor].join('-');
  const color: string = ['open-proj-txt',txtColor].join('-');
  
  return (
    <button
      type="button"
      className={[color, background].join(' ')}
      {...props}
    >
      {label}
    </button>
  );
};
