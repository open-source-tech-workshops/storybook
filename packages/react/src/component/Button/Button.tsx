import React from "react";
import ButtonConstants from "./ButtonConstants";
import ButtonType from "./Type";
import "./Button.css";

const Button: React.FC<ButtonType> = ({
  color = "blue",
  label = "button",
  clickHandler,
}) => {
  const buttonClass = `twk-button twk-button-color-${color}`;
  return (
    <button
      className={buttonClass}
      onClick={() => {
        if (clickHandler) clickHandler();
      }}
      data-testid={ButtonConstants.ButtonTestId.toString()}
    >
      <span className="twk-text">{label}</span>
    </button>
  );
};

export default Button;
