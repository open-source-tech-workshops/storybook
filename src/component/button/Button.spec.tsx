import * as React from 'react';
import { act, fireEvent, render, RenderResult, screen } from "@testing-library/react";
import { Button } from './Button';
import { ReactElement, ReactNode } from 'react';
import '@testing-library/jest-dom/extend-expect';

const labelText: string = 'Click Me';
describe('Test Auth Webapp entry', function () {
    it('should display Button with default value', ()=>{
        const component: ReactElement = <Button label='Click Me'/>;
        render(component);
        expect(screen.getByText(labelText, {exact: true})).toMatchSnapshot();
        expect(screen.getByRole('button')).toHaveTextContent(labelText);
    })
    
    it('should display Button with text color and background color', ()=>{
        const component: ReactElement = <Button label='Click Me2' txtColor='red' bgColor='white'/>;
        render(component);
        expect(screen.getByText('Click Me2')).toBeInTheDocument();
        expect(screen.getByText('Click Me2')).toHaveClass('open-proj-btn-white');
        expect(screen.getByText('Click Me2')).toHaveClass('open-proj-txt-red');

    })
   

});