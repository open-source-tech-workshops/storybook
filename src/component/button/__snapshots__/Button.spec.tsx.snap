// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Auth Webapp entry should display Button with default value 1`] = `
<button
  class="open-proj-txt- open-proj-btn-"
  type="button"
>
  Click Me
</button>
`;
